cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -251,89 +254,103 @@ protected void processStage(PhaseSpec phaseSpec, String sourcePluginType, JavaSp
 
     StageSpec stageSpec = pipelinePhase.getStage(stageName);
     String pluginType = stageSpec.getPluginType();
+    JavaSparkContext jsc = getSparkContext();
 
-    // don't want to do an additional filter for stages that can emit errors,
-    // but aren't connected to an ErrorTransform
-    // similarly, don't want to do an additional filter for alerts when the stage isn't connected to
-    // an AlertPublisher
-    boolean hasErrorOutput = false;
-    boolean hasAlertOutput = false;
-    Set<String> outputs = pipelinePhase.getStageOutputs(stageName);
-    for (String output : outputs) {
-      String outputPluginType = pipelinePhase.getStage(output).getPluginType();
-      //noinspection ConstantConditions
-      if (ErrorTransform.PLUGIN_TYPE.equals(outputPluginType)) {
-        hasErrorOutput = true;
-      } else if (AlertPublisher.PLUGIN_TYPE.equals(outputPluginType)) {
-        hasAlertOutput = true;
+    try {
+      String pluginName = stageSpec.getPlugin().getName();
+      String artifactInfo = "";
+      if (stageSpec.getPlugin().getArtifact() != null) {
```

Comment:
Well, if the problem is this null, just don't call `.toString()' explicitly. Implicit "+" will work with null values. You need `if` only if you want to check if getPlugin() is null